src/Psalm/Internal/Analyzer/Statements/Expression/CastAnalyzer.php

```diff
@@ -352,7 +778,14 @@ public static function castStringAttempt(
                 continue;
             }
 
+            if ($atomic_type instanceof TBool
+            ) {
+                $valid_strings[] = new TLiteralString('1');
+                continue;
+            }
+
             if ($atomic_type instanceof TMixed
+                || $atomic_type instanceof TClosedResource
```

Comment:
Does PHP ever give a warning for this? Do we need to add an issue in some cases?

Reply:
According to [this](https://www.php.net/manual/en/language.types.string.php#language.types.string.casting):

> Resources are always converted to strings with the structure "Resource id #​1", where 1 is the resource number assigned to the resource by PHP at runtime. While the exact structure of this string should not be relied on and is subject to change, it will always be unique for a given resource within the lifetime of a script being executed (ie a Web request or CLI process) and won't be reused.

so this could probably be non-empty-string.

Reply:
> And resource/closed-resource TIntRange(1, null) for typecasts to (int) too

I see [this](https://www.php.net/manual/en/language.types.integer.php#language.types.integer.casting) which says:

> If a resource is converted to an int, then the result will be the unique resource number assigned to the resource by PHP at runtime.

I don't see anything guaranteeing it's positive, and I found [this](https://stackoverflow.com/a/6946116) saying non-positive numbers aren't expected, but they are possible. @orklah what do you think about that? Maybe `int` is safer here since I'm not sure if a non-positive value is considered a bug by PHP?